core/dbt/contracts/graph/unparsed.py

```diff
@@ -736,10 +738,54 @@ def normalize_date(d: Optional[datetime.date]) -> Optional[datetime.datetime]:
     return dt
 
 
+class UnitTestFormat(StrEnum):
+    CSV = "csv"
+    Dict = "dict"
+
+
+class UnitTestFixture:
+    @property
+    def format(self) -> UnitTestFormat:
+        return UnitTestFormat.Dict
+
+    @property
+    def rows(self) -> Union[str, List[Dict[str, Any]]]:
+        return []
+
+    def get_rows(self) -> List[Dict[str, Any]]:
+        if self.format == UnitTestFormat.Dict:
+            assert isinstance(self.rows, List)
+            return self.rows
+        elif self.format == UnitTestFormat.CSV:
+            assert isinstance(self.rows, str)
+            dummy_file = StringIO(self.rows)
+            reader = csv.DictReader(dummy_file)
+            rows = []
+            for row in reader:
+                rows.append(row)
+            return rows
+
+    @classmethod
+    def validate_fixture(cls, fixture, fixture_type, test_name):
```

Comment:
What is the utility of this as a classmethod given it is passed an instance of itself? Perhaps a regular method would be more straightforward. Also, types on this signature would be great!

Reply:
Oh, right. I originally created it as an override of the 'validate' class method, but that didn't work because it had to have a different signature.